File: 06.帳票WEB API(旧式WCF+SOAP)－クラウド連携中/CS/SoapService/IService1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.ServiceModel.Web;
using System.Text;

namespace WCFServiceWebRole1
{
    // メモ: [リファクター] メニューの [名前の変更] コマンドを使用すると、コードと config ファイルの両方で同時にインターフェイス名 "IService1" を変更できます。
    [ServiceContract]
    public interface IService1
    {

        [OperationContract]
        string GetData(int value);

        [OperationContract]
        CompositeType GetDataUsingDataContract(CompositeType composite);

        // TODO: ここにサービス操作を追加します。
        [OperationContract]
        byte[] getReports単純なサンプル();
        [OperationContract]
        byte[] getReports10の倍数();
        [OperationContract]
        byte[] getReports見積書();
        [OperationContract]
        byte[] getReports郵便番号();
        [OperationContract]
        byte[] getReports広告();
        [OperationContract]
        byte[] getReports請求書();
        [OperationContract]
        byte[] getReports商品一覧();


    }


    // サービス操作に複合型を追加するには、以下のサンプルに示すようにデータ コントラクトを使用します。
    [DataContract]
    public class CompositeType
    {
        bool boolValue = true;
        string stringValue = "Hello ";

        [DataMember]
        public bool BoolValue
        {
            get { return boolValue; }
            set { boolValue = value; }
        }

        [DataMember]
        public string StringValue
        {
            get { return stringValue; }
            set { stringValue = value; }
        }
    }
}

File: 06.帳票WEB API(旧式WCF+SOAP)－クラウド連携中/VB/SoapService/IService1.vb
' メモ: コンテキスト メニューの [名前の変更] コマンドを使用すると、コードと config ファイルの両方で同時にインターフェイス名 "IService1" を変更できます。
<ServiceContract()>
Public Interface IService1

    <OperationContract()>
    Function GetData(ByVal value As Integer) As String

    <OperationContract()>
    Function GetDataUsingDataContract(ByVal composite As CompositeType) As CompositeType

    ' TODO: ここにサービス操作を追加します。
    <OperationContract()>
    Function getReports単純なサンプル() As Byte()
    <OperationContract()>
    Function getReports10の倍数() As Byte()
    <OperationContract()>
    Function getReports見積書() As Byte()
    <OperationContract()>
    Function getReports郵便番号() As Byte()
    <OperationContract()>
    Function getReports広告() As Byte()
    <OperationContract()>
    Function getReports請求書() As Byte()
    <OperationContract()>
    Function getReports商品一覧() As Byte()

End Interface

' サービス操作に複合型を追加するには、以下のサンプルに示すようにデータ コントラクトを使用します。

<DataContract()>
Public Class CompositeType

    <DataMember()>
    Public Property BoolValue() As Boolean

    <DataMember()>
    Public Property StringValue() As String

End Class
